cmd/complex/main.go

```diff
@@ -25,28 +26,41 @@ func newSSE() *sse.Server {
 	rp, _ := sse.NewValidReplayer(time.Minute*5, true)
 	rp.GCInterval = time.Minute
 
+	logFunc := func(r *http.Request) *slog.Logger {
+		l, err := getLogger(r)
+		if err != nil {
+			return nil
+		}
+		return l
+	}
+
 	return &sse.Server{
 		Provider: &sse.Joe{Replayer: rp},
-		Logger:   logger{log.New(os.Stderr, "", 0)},
-		OnSession: func(s *sse.Session) (sse.Subscription, bool) {
-			topics := s.Req.URL.Query()["topic"]
+		Logger:   logFunc,
```

Comment:
Though having a comment hinting at external helper usage is helpful. I'd probably add that comment here, as this knowledge is most relevant when initializing the `sse.Server`.
```suggestion
		// If you're using a third-party logging library and have ensured that this library
		// adds the logger to the request context, use here its own helper function
		// to retrieve a logger from context.
		Logger:   getLogger,
```
I've also changed `logFunc` to `getLogger`, assuming the other proposed change to the signature of `getLogger`.

Reply:
I added the comment and just made the logFunc inline in the struct initialization. It seems totally readable that way and is closer to what was originally there.